openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -2647,6 +2666,88 @@ TCN_IMPLEMENT_CALL(jboolean, SSLContext, setCurvesList0)(TCN_STDARGS, jlong ctx,
     return ret == 1 ? JNI_TRUE : JNI_FALSE;
 }
 
+TCN_IMPLEMENT_CALL(jint, SSLContext, addCertificateCompressionAlgorithm0)(TCN_STDARGS, jlong ctx, jint direction, jint algorithmId, jobject algorithm) {
+    tcn_ssl_ctxt_t *c = J2P(ctx, tcn_ssl_ctxt_t *);
+    TCN_CHECK_NULL(c, ctx, 0);
+    if (algorithm == NULL) {
+        tcn_ThrowIllegalArgumentException(e, "Compression algorithm may not be null");
```

Comment:
We need to return early.

```suggestion
        tcn_ThrowIllegalArgumentException(e, "Compression algorithm may not be null");
        return 0;
```

Reply:
fixed